scripts/service-catalog.py

```diff
@@ -0,0 +1,270 @@
+        difference = set(f1).difference(f2)
+
+    print difference
+    if difference == diff_set:
+        print False,old_template_path
+        return (False,old_template_path)
+    else:
+        print True,new_template_path
+        return (True,new_template_path)
+
+def get_latest_version_template(ser_cat_clt_conn,latest_version_id,product_id):
+    response = ser_cat_clt_conn.describe_provisioning_artifact(ProvisioningArtifactId=latest_version_id,ProductId=product_id)
+    print "latest template = {}".format(response['Info']['TemplateUrl'])
+    return response['Info']['TemplateUrl']
+
+
+
+def portfolio(ser_cat_clt_conn,region):
+    """To create portfolio
+    """
+    portfolio_dict = ser_cat_clt_conn.list_portfolios(PageSize=20)
+    if portfolio_dict['PortfolioDetails']==[]:
+        print "creating portfolio {} in region {}".format(PORTFOLIO_NAME,region)
+        portfolio_id = create_portfolio(ser_cat_clt_conn,PORTFOLIO_NAME,region)
+    elif portfolio_dict['PortfolioDetails'] != []:
+        for  portfolio in portfolio_dict['PortfolioDetails']:
+            if portfolio['DisplayName']==PORTFOLIO_NAME:
+                print "portfolio {} already exist in region {}".format(PORTFOLIO_NAME,region)
+                portfolio_id = portfolio['Id']
+                break
+        else:
+            print "creating portfolio {} in region {}".format(PORTFOLIO_NAME,region)
+            portfolio_id = create_portfolio(ser_cat_clt_conn,PORTFOLIO_NAME,region)
+
+    return portfolio_id
+
+def main(temp_s3_url,product_name,conn,product_template,portfolio_id):
+    ser_cat_clt_conn = conn[0]
+    region = conn[1]
+    client_s3 = conn[2]
+
+    """TO create product
+    """
+    response = ser_cat_clt_conn.search_products()
+    for product in response['ProductViewSummaries']:
+        if product['Name'] == product_name:
+            product_id =  product['ProductId']
+            version_response = ser_cat_clt_conn.describe_product(Id=product_id)
+            tdict= {}
+            vdict= {}
+            for version in  version_response['ProvisioningArtifacts']:
+                tdict[time.mktime(version['CreatedTime'].timetuple())]=version['Id']
+                vdict[time.mktime(version['CreatedTime'].timetuple())]=version['Name']
+
+            latest_version_id= tdict[max(tdict.keys())]
+            latest_version_name= vdict[max(vdict.keys())]
+            print latest_version_id
+            print latest_version_name
+            template_utl = get_latest_version_template(ser_cat_clt_conn,latest_version_id,product_id)
+            print "product_template=",product_template
+
+            comp_status = compare_templates(conn,template_utl,product_name,product_template)
+
+            # create version of product if template changed
+            if comp_status[0] == True:
+                global VERSION
+                VERSION = "v"+str(float(latest_version_name.split("v")[1])+.1)
```

Comment:
so how does versioning scheme work? is it like v1.0 v1.1 ..? I was thinking we could do something like v1.0, v2.0 etc

Reply:
it is like v1.0 then v1.1